src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -177,6 +177,32 @@ export default class ReimbursementRequestService {
     return updatedReimbursementRequest;
   }
 
+  /**
+   * Soft-deletes the given reimbursement request
+   *
+   * @param requestId the request to be deleted
+   * @param deleter the user deleting the request
+   */
+  static async deleteReimbursementRequest(requestId: string, deleter: User): Promise<Reimbursement_Request> {
+    const request = await prisma.reimbursement_Request.findUnique({
+      where: { reimbursementRequestId: requestId }
+    });
+
+    if (!request) throw new NotFoundException('Reimbursement Request', requestId);
+    if (request.recipientId !== deleter.userId)
+      throw new AccessDeniedException(
+        'You do not have access to delete this reimbursement request, only the creator can delete a reimbursement request'
+      );
+    if (request.dateDeleted) throw new DeletedException('Reimbursement Request', requestId);
+
```

Comment:
we should also throw an exception if the request has been approved by the finance team (see https://github.com/Northeastern-Electric-Racing/FinishLine/issues/493, View Reimbursement Request section)